crates/oxc_linter/src/rules/jsdoc/check_property_names.rs

```diff
@@ -80,13 +77,22 @@ impl Rule for CheckPropertyNames {
                 let type_name = name_part.parsed();
 
                 // Check duplicated
-                if seen.contains(&type_name) {
-                    ctx.diagnostic(CheckPropertyNamesDiagnostic::Duplicated(
-                        name_part.span,
-                        type_name.to_string(),
+                if let Some(duplicated_span) = seen.get(&type_name) {
+                    ctx.diagnostic(miette!(
+                        severity = Severity::Warning,
+                        labels = [duplicated_span, &name_part.span]
+                            .iter()
+                            .map(|span| LabeledSpan::at(
+                                (span.start as usize)..(span.end as usize),
+                                "Duplicated property".to_string(),
+                            ))
+                            .collect::<Vec<_>>(),
+                        help = format!("@property `{type_name}` is duplicated on the same block."),
+                        "eslint-plugin-jsdoc(check-property-names): Duplicate @property found."
```

Comment:
Oh, I thought there are more than 2.

For only 2 spans, e.g.:

```
    #[derive(Debug, Error, Diagnostic)]
    #[error("Duplicated export '{0}'")]
    #[diagnostic()]
    struct DuplicateExport(
        CompactStr,
        #[label("Export has already been declared here")] Span,
        #[label("It cannot be redeclared here")] Span,
    );
```

Reply:
Thanks for the suggestions!

> collect all duplicates and emit them all at once at the end.

I'll go this way. 👉🏻  => https://github.com/oxc-project/oxc/pull/2989/commits/e253a071515b1044cf7b14023f47b349cd969732